appinventor/blocklyeditor/src/versioning.js

```diff
@@ -3041,18 +3041,22 @@ Blockly.Versioning.AllUpgradeMaps =
   }, // End Spinner upgraders
 
   "Spreadsheet": {
-    2: [
+    3: [
       Blockly.Versioning.changeMethodName("Spreadsheet", "ReadCol", "ReadColumn"),
       Blockly.Versioning.changeMethodName("Spreadsheet", "WriteCol", "WriteColumn"),
       Blockly.Versioning.changeMethodName("Spreadsheet", "AddCol", "AddColumn"),
       Blockly.Versioning.changeMethodName("Spreadsheet", "RemoveCol", "RemoveColumn"),
+      Blockly.Versioning.changeMethodName("Spreadsheet", "AddSheet", "noUpgrade"),
+      Blockly.Versioning.changeMethodName("Spreadsheet", "DeleteSheet", "noUpgrade"),
       Blockly.Versioning.changeEventName("Spreadsheet", "GotColData", "GotColumnData"),
       Blockly.Versioning.changeEventName("Spreadsheet", "FinishedWriteCol", "FinishedWriteColumn"),
       Blockly.Versioning.changeEventName("Spreadsheet", "FinishedAddCol", "FinishedAddColumn"),
       Blockly.Versioning.changeEventName("Spreadsheet", "FinishedRemoveCol", "FinishedRemoveColumn"),
       Blockly.Versioning.changeEventParameterName("Spreadsheet", "GotFilterResult", "return_rows", "returnRows"),
       Blockly.Versioning.changeEventParameterName("Spreadsheet", "GotFilterResult", "return_data", "returnData"),
-      Blockly.Versioning.changeEventParameterName("Spreadsheet", "GotColumnData", "colDataList", "columnData")
+      Blockly.Versioning.changeEventParameterName("Spreadsheet", "GotColumnData", "colDataList", "columnData"),
+      Blockly.Versioning.changeEventName("Spreadsheet", "FinishedAddSheet", "noUpgrade"),
+      Blockly.Versioning.changeEventName("Spreadsheet", "FinishedDeleteSheet", "noUpgrade")
```

Comment:
Don't change existing version numbers. Instead, add the a new version number entry and specify that upgrader is "noUpgrade". See the Spinner class above as an example.